hazelcast/src/test/java/com/hazelcast/cluster/SplitBrainHandlerTest.java

```diff
@@ -750,6 +753,58 @@ public void stateChanged(LifecycleEvent event) {
         assertMasterAddress(getAddress(hz3), hz1, hz2, hz3);
     }
 
+    @Test
+    // https://github.com/hazelcast/hazelcast/pull/24830
+    public void testBlacklistsConsideredAndRecovered_WhenUsingTcpIpJoiner() {
+        Config configA = createSimpleTcpIpConfig("cluster_A");
+        Config configB = createSimpleTcpIpConfig("cluster_B");
+
+        // Start 2 clusters, with 2 members each
+        HazelcastInstance a1 = Hazelcast.newHazelcastInstance(configA);
+        HazelcastInstance a2 = Hazelcast.newHazelcastInstance(configA);
+        HazelcastInstance b1 = Hazelcast.newHazelcastInstance(configB);
+        HazelcastInstance b2 = Hazelcast.newHazelcastInstance(configB);
+
+        // Fetch the port of a2
+        Address a2Address = getNode(a2).getThisAddress();
+
+        // Confirm that A2 is blacklisted by B1 after a SplitBrainHandler run (due to cluster mismatch)
+        Joiner joinerB1 = getNode(b1).getJoiner();
+        assertTrueEventually("Node A2 has not been blacklisted by Node B1!",
+                () -> assertTrue(joinerB1.isBlacklisted(a2Address)), 6);
+
+        // Confirm that A2's address is not returned in the list of filtered addresses used in SplitBrainHandler
+        assertFalse(((TcpIpJoiner) joinerB1).getFilteredPossibleAddresses().contains(a2Address));
+
+        // Shutdown A2, and start B3, which should be on the same address/port
+        a2.shutdown();
+        HazelcastInstance b3 = Hazelcast.newHazelcastInstance(configB);
+
+        // Confirm b3 is running on the address/port previously occupied by a2
+        Address b3Address = getNode(b3).getThisAddress();
+        assertEquals(a2Address, b3Address);
+
+        // Confirm B3 joins the cluster with B1/B2
+        assertClusterSizeEventually(3, b1, b2, b3);
+
+        // Confirm that B3's address is not blacklisted on B1, allowing SplitBrainHandler checks
+        assertTrueEventually("Node B3 is still blacklisted by Node B1!",
+                () -> assertFalse(joinerB1.isBlacklisted(b3Address)), 6);
+    }
+
+    private Config createSimpleTcpIpConfig(String clusterName) {
+        Config config = new Config();
```

Comment:
you can initialize this config with `smallInstanceConfig()` so it is lighter on resources (fewer threads created, jet & metrics disabled)

Reply:
Thanks for the tip, good point! Done ⭐ https://github.com/hazelcast/hazelcast/pull/24830/commits/e861df9d778588267bc977c37c6a971a81735e44